############ the following command are aplied when the fasta file generated by transdecoder are loaded into R and fileter the only those transcripts which has "ATG" as strat
############ and (TAA|TAG|TGA) as stop codons ##############################################


# Loading library seqinr
library(seqinr)

# Loading your input file
fa <- read.fasta("transdecoder_input.fa")

# converting the character format into a string format for regex operation
sequences_1 <- sapply(fa, c2s)

# Retaining only sequences which has atg as start codon and stop codon as (taa|tga|tag)

atg_stop_codons=grep("^atg.*(taa|tga|tag)$",sequences_1,perl=T,value=T)

#Converting string format into character format
start_stop=sapply(atg_stop_codons,s2c)

# Getting sequences and gene names from fa object
genes <- sapply(strsplit(names(start_stop), "\\|"), function(v) {return(v[1])})
sequences_2 <- sapply(start_stop, c2s)

# Extracting longest transcript for each gene
filtered_seq <- tapply(sequences_2, genes, function(v) {return(v[which(nchar(v)==max(nchar(v)))])})

# Writing output file
write.fasta(filtered_seq , names(filtered_seq), file="output.fa")